src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1620,97 +1236,6 @@ public SolutionState WithAnalyzerReferences(IReadOnlyList<AnalyzerReference> ana
-        /// </summary>
-        public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, CancellationToken cancellationToken)
-        {
-            try
-            {
-                var allDocumentIds = GetRelatedDocumentIds(documentId);
-                using var _ = ArrayBuilder<(DocumentState, SyntaxTree)>.GetInstance(allDocumentIds.Length, out var builder);
-
-                foreach (var currentDocumentId in allDocumentIds)
-                {
-                    var document = this.GetRequiredDocumentState(currentDocumentId);
-                    builder.Add((document, document.GetSyntaxTree(cancellationToken)));
-                }
-
-                using (this.StateLock.DisposableWait(cancellationToken))
-                {
-                    // in progress solutions are disabled for some testing
-                    if (Services.GetService<IWorkspacePartialSolutionsTestHook>()?.IsPartialSolutionDisabled == true)
-                    {
-                        return this;
-                    }
-
-                    SolutionState? currentPartialSolution = null;
-                    _latestSolutionWithPartialCompilation?.TryGetTarget(out currentPartialSolution);
-
-                    var reuseExistingPartialSolution =
-                        currentPartialSolution != null &&
-                        (DateTime.UtcNow - _timeOfLatestSolutionWithPartialCompilation).TotalSeconds < 0.1 &&
-                        _documentIdOfLatestSolutionWithPartialCompilation == documentId;
-
-                    if (reuseExistingPartialSolution)
-                    {
-                        SolutionLogger.UseExistingPartialSolution();
-                        return currentPartialSolution!;
-                    }
-
-                    var newIdToProjectStateMap = _projectIdToProjectStateMap;
-                    var newIdToTrackerMap = _projectIdToTrackerMap;
-
-                    foreach (var (doc, tree) in builder)
-                    {
-                        // if we don't have one or it is stale, create a new partial solution
-                        var tracker = this.GetCompilationTracker(doc.Id.ProjectId);
-                        var newTracker = tracker.FreezePartialStateWithTree(this, doc, tree, cancellationToken);
-
-                        Contract.ThrowIfFalse(newIdToProjectStateMap.ContainsKey(doc.Id.ProjectId));
-                        newIdToProjectStateMap = newIdToProjectStateMap.SetItem(doc.Id.ProjectId, newTracker.ProjectState);
-                        newIdToTrackerMap = newIdToTrackerMap.SetItem(doc.Id.ProjectId, newTracker);
-                    }
-
-                    currentPartialSolution = this.Branch(
-                        idToProjectStateMap: newIdToProjectStateMap,
-                        projectIdToTrackerMap: newIdToTrackerMap,
-                        dependencyGraph: CreateDependencyGraph(ProjectIds, newIdToProjectStateMap));
```

Comment:
a complex case where we are building up new maps for the project state and tracker state at teh same time.  this will be worth looking closely at in SolutionCompilationState.